doc/design/service-discovery.md

```diff
@@ -0,0 +1,316 @@
+# Service Discovery
+
+## Summary and problem description
+
+Service discovery over IPC channel e.g. message queue or UNIX domain socket is not performant since larger data is
+transferred, which can lead to transmission of several frames. If lots of services are discovered at high-frequency
+e.g. at startup the IPC channel can become a bottleneck. Furthermore, a lesson learned from the iceoryx development
+so far is, that the IPC channel should just be used for the earliest communication during startup and not for the
+creation of objects inside the shared memory (see [this issue](https://github.com/eclipse-iceoryx/iceoryx/issues/611)).
```

Comment:
I agree.
The lock-free structures (queues and management of chunks for publishers and subscribers) are an issue but the queue should be decently fast even under contention. I assume most of the time is spend in calls on different abstraction layers for the message passing and managing the chunk cache.

This can be avoided with a more direct approach (all of them are IPC in a way) as @elBoberido said.